pkg/auth/dependency/newinteraction/nodes/update_identity_end.go

```diff
@@ -0,0 +1,74 @@
+package nodes
+
+import (
+	"errors"
+
+	"github.com/authgear/authgear-server/pkg/auth/dependency/identity"
+	"github.com/authgear/authgear-server/pkg/auth/dependency/newinteraction"
+)
+
+func init() {
+	newinteraction.RegisterNode(&NodeUpdateIdentityEnd{})
+}
+
+type EdgeUpdateIdentityEnd struct {
+	IdentitySpec *identity.Spec
+}
+
+func (e *EdgeUpdateIdentityEnd) Instantiate(ctx *newinteraction.Context, graph *newinteraction.Graph, rawInput interface{}) (newinteraction.Node, error) {
+	userID := graph.MustGetUserID()
+	identityType := e.IdentitySpec.Type
+	identityID := graph.MustGetUpdateIdentityID()
```

Comment:
`UpdateIdentityID` seems to be a state local to the `UpdateIdentityBegin/End` pair of nodes. I'd prefer to pass local states through fields in edges, instead of retrieving globally from the graph.

Reply:
They are separated by a generic edge EdgeUseIdentityLoginID, which makes it non-trivial to pass local state.

Reply:
Hmm... I suppose we can review it in future whether having a generic UseIdentity node with three modes is the best way forward.